maven/lib/dependabot/maven/new_version.rb

```diff
@@ -0,0 +1,68 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "dependabot/maven/version_parser"
+require "dependabot/version"
+require "dependabot/utils"
+
+# See https://maven.apache.org/pom.html#Version_Order_Specification for details.
+
+module Dependabot
+  module Maven
+    class NewVersion
+      extend T::Sig
+      extend T::Helpers
+
+      PRERELEASE_QUALIFIERS = T.let([
+        Dependabot::Maven::VersionParser::ALPHA,
+        Dependabot::Maven::VersionParser::BETA,
+        Dependabot::Maven::VersionParser::MILESTONE,
+        Dependabot::Maven::VersionParser::RC,
+        Dependabot::Maven::VersionParser::SNAPSHOT
+      ].freeze, T::Array[Integer])
+
+      sig { returns(Dependabot::Maven::TokenBucket) }
+      attr_accessor :token_bucket
+
+      sig { params(version: T.nilable(String)).returns(T::Boolean) }
+      def self.correct?(version)
+        return false if version.nil? || version.empty?
+
+        Dependabot::Maven::VersionParser.parse(version.to_s).to_a.any?
+      end
+
+      sig { params(version: T.nilable(String)).void }
```

Comment:
is there any way to avoid Nilables as much as possible?

Reply:
I've removed this one